Neos.Neos/Classes/Domain/Workspace/WorkspaceFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Neos.Neos package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\Workspace;
+
+use Neos\ContentRepository\Core\ContentRepository;
+use Neos\ContentRepository\Core\Projection\Workspace\Workspace as ContentRepositoryWorkspace;
+use Neos\ContentRepository\Core\SharedModel\ContentRepository\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Neos' factory for its own workspace instances
+ *
+ * @api
+ */
+#[Flow\Scope('singleton')]
+final readonly class WorkspaceFactory
```

Comment:
i guess it would be a good place to add a `createForUser` method or the like?

Currently we have logic for this in 1000 places. But to ease reviewing this might really be done as followup.... the only thing is to consider if side by side `create` is too generic then and should rather be `createForName` but idk.

Reply:
I've renamed the method to be similar to other factory methods we have (i.e. createFromParameterAAndParameterB)